### PR TITLE
Remove dependency from core module to paper module

### PR DESCRIPTION
**Pull request by sulume**
I think it's a little bit strange if a core module depends on an advanced module because that leads users of this module to install a lot of dependent modules unnecessarily by bower.
This request contains a removal of a dependency to 'paper-fab' and two extra fixes on demo as separated commits. Please feel free to cherrypick or reject any of them.

### CODE REVIEW

**Comment by ebidel** [comment]
This dep is only for the demo. @nevir @frankiefu  what is the updated "style guide" for including demo dep in elements?

**Comment by ebidel** [comment]
Thanks for the PR! We're changing the folder structure of demos for 0.8. Would you mind updating this PR to use a demo/ folder and `devDependencies` in bower.json, keep paper-fab.

Example: https://github.com/nevir/seed-element/tree/0.8-preview

**Comment by sulume** [comment]
Done. Would you like to take this into some other branch like 0.8-preview?

File: demo/index.html [hunk 1]
```diff
@@ -86,7 +86,8 @@
     }
 
     core-tooltip.fancy:hover::shadow .core-tooltip,
-    core-tooltip.fancy:focus::shadow .core-tooltip {
+    core-tooltip.fancy:focus::shadow .core-tooltip,
+    core-tooltip.fancy[show]::shadow .core-tooltip {
```
Comment (ebidel) [inline demo/index.html]: This change seems unrelated...?

Comment (sulume) [inline demo/index.html]: This is a bugfix for the demo script of [Toggle all] button that is unrelated to the dependency removal. I'll exclude the unrelated changes 15ec8de, dff0c7f if you prefer to merge a single-purpose patch.

Comment (ebidel) [inline demo/index.html]: Got it

**Comment by ebidel** [comment]
LGTM. Thanks!
